### DISCUSSION TITLE
databricks ssl issue when using airflow 2.3.2

### DISCUSSION THREAD

**Discussion by Achinm** [apache/airflow#24850] [Q&A]
I have airflow 2.3.2 version installed on local machine and databricks on Azure (databricks is authenticated and configured with Company AD on company account). I am trying to trigger these cloud Databricks' jobs from local airflow DAGs, but I am getting below ssl error. I have wsl installed ubuntu version 18.04 lts through which I have configured airflow and trying to do all the above. I tried creating a VM on Azure (Ubuntu). I am getting the same error there as well. So this error cannot be due to OS dependencies. I am able to run the databricks jobs using the cli and postman by disabling the ssl checks using the same local system and VM. 

PS: I have created the Databricks connection in the Airflow UI connection pane and trying to use it with DatabricksRunNowOperator().

   **raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='adb-1091324782758285.5.azuredatabricks.net', port=443): Max retries exceeded with url: /api/2.1/jobs/run-now (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1131)')))**

Please advise what is the exact solution for this problem as this might hamper us in using the product any further. Some quick help will be highly cherished.

**Comment by potiuk** [comment]
This looks like a problem that you should handle with your machines/deployment/enviromment

You have some problem with local ca-certifcates. Google search returns this as first result: 
 https://stackoverflow.com/questions/52805115/certificate-verify-failed-unable-to-get-local-issuer-certificate 

So my suggestion is to follow ths and maybe few other results of google search.

When it come to cherishing. Highly appreciate. Since yoy expect an "exact solution" - I guess you are used to paid support. You do realise airflow is a free software and it does not have "on-demand support", so my question is - how much would you will to pay to expect "exact solution"? Is "cherishing" and offer to pay for support ? I can recommend a few companies that provide paid support for airflow and I can provide you with some leads -just name your price.

**Reply by Achinm** [reply]
Hi Jarek, 

Thanks for your reply, I greatly appreciate it.

I went through this blog during my analysis and this gives an answer to chnage or review the certifi library (which I already tried) but to no avail for me as of now. I understand that the DatabricksRunNow Operator would probably use the latest or the installed version of this library to fire requests. Is my understanding correct?

Hence, I already tried this by changing the library versions but both my environments WSL Ubuntu on windows on local machine + Ubuntu on VM give me the same error. Are others facing this issue and what are the steps taken to resolve it ?

All the DAG requests are getting failed for this ssl issue only.

**Reply by potiuk** [reply]
I think you need to look deeper. This is not a problem of Airlfow but underlying operating system and depioyment - make syre they are using the latest versions (guessing now). This is really in your hands to solve it. If you can't solve it - you can also raise your issue to databricks support. The operator uses databricks libraries that they maintain it also @alexott maybe you can help with the question.

**Reply by alexott** [reply]
@Achinm Can you confirm that you don't have web proxy between databricks & your machine? Such problems usually occur when web proxy scans HTTPS traffic & generate a new certificate signed with the proxy's CA. In this case, usual practice is put that CA into the `certifi` chain

**Reply by potiuk** [reply]
Or you have someone (government ?)  snooping at your traffic :P
